support/controlplane-component/defaults.go

```diff
@@ -83,6 +83,7 @@ func (c *controlPlaneWorkload[T]) setDefaultOptions(cpContext ControlPlaneContex
 
 	podTemplateSpec := c.workloadProvider.PodTemplateSpec(workloadObj)
 	enforceVolumesDefaultMode(&podTemplateSpec.Spec)
+	enforceReadOnlyRootFilesystem(&podTemplateSpec.Spec)
```

Comment:
This call should be moved after konnectivity, token-minter and availabilityProber containers are injected. i.e. after defaults.go#L122.
You wouldn't need to duplicate this logic in konnecitivty-cotainer.go and token-minter-container.go then